### PR TITLE
Aa v5 rebase

### PR DESCRIPTION
**Pull request by livingrockrises**
updating the interfaces IAccount and IPaymaster and corresponding changes in the account and paymaster contracts.

validateUserOp and validatePaymasterUserOp returns validationData now instead of sigTimeRange
( validation data is : =  time range validAfter, validBefore, returned signature check (can have signatureAgregator) )

(list of diffs below...

IAggregatedAccount is obsolete
Helpers.sol is a new contract 
IAccount https://www.diffchecker.com/gczWisyo/
IAggregator https://www.diffchecker.com/UXE0DhA1/ 
IEntryPoint https://www.diffchecker.com/jVsLC0M8/  
IPaymaster https://www.diffchecker.com/PZkg9V2C/
IStakeManager.sol https://www.diffchecker.com/YklFedVo/ 
UserOperation https://www.diffchecker.com/Dnoexj8Y/
BaseAccount https://www.diffchecker.com/GQ0KJ49x/
BasePaymaster https://www.diffchecker.com/lMRBx0zw/
EntryPoint https://www.diffchecker.com/EdNVXwsM/
SenderCreator https://www.diffchecker.com/tmNZSXRN/
StakeManager https://www.diffchecker.com/TD18wrh2/
SimpleAccount https://www.diffchecker.com/MtSLoWaW/
SimpleAccountFactory https://www.diffchecker.com/LHMarUuB/
VerifyingPaymaster https://www.diffchecker.com/bGLM1TAL/
)

** updated entry point tests as per the new contract

### CODE REVIEW

File: contracts/smart-contract-wallet/BaseSmartAccount.sol [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {IEntryPoint} from "@account-abstraction/contracts/interfaces/IEntryPoint
 import {UserOperationLib, UserOperation} from "@account-abstraction/contracts/interfaces/UserOperation.sol";
 import {Enum} from "./common/Enum.sol";
 import {BaseSmartAccountErrors} from "./common/Errors.sol";
+import "@account-abstraction/contracts/core/Helpers.sol";
```
Comment (AmanRaj1608) [inline contracts/smart-contract-wallet/BaseSmartAccount.sol:13]: should use the named imports as per the QA done

File: contracts/smart-contract-wallet/aa-4337/test/TestSignatureAggregator.sol [hunk 2]
```diff
@@ -4,26 +4,28 @@ pragma solidity 0.8.17;
...
-            uint256 nonce = userOps[i].nonce;
+        uint sum = 0;
+        for (uint i = 0; i < userOps.length; i++) {
+            uint nonce = userOps[i].nonce;
             sum += nonce;
         }
-        require(signature.length == 32, "TestSignatureValidator: sig must be uint256");
-        (uint256 sig) = abi.decode(signature, (uint256));
+        require(signature.length == 32, "TestSignatureValidator: sig must be uint");
+        (uint sig) = abi.decode(signature, (uint));
```
Comment (AmanRaj1608) [inline contracts/smart-contract-wallet/aa-4337/test/TestSignatureAggregator.sol:24]: same for these changes, had decided to use the uint256

Comment (livingrockrises) [reply contracts/smart-contract-wallet/aa-4337/test/TestSignatureAggregator.sol:24]: its just test contract from infinitism samples so imo shouldn't matter

File: contracts/smart-contract-wallet/BaseSmartAccount.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/smart-contract-wallet/BaseSmartAccount.sol:13]: would have been too many named imports as this file is a library

**Review by AmanRaj1608 — APPROVED** [review]
LGTM!

**Pull request merged**